reviewrot/base.py

```diff
@@ -0,0 +1,101 @@
+from collections import OrderedDict
+
+
+class BaseService(object):
+
+    def format_duration(self, rel_diff):
+        """
+        Formats the duration the review request is pending for
+        Args:
+            rel_diff (relativedelta): duration the review request is pending
+                                      for
+
+        Returns:
+            a string of duration the review request is pending for
+        """
+        time_dict = OrderedDict([
+            ('year', rel_diff.years),
+            ('month', rel_diff.months),
+            ('day', rel_diff.days),
+            ('hour', rel_diff.hours),
+            ('minute', rel_diff.minutes),
+        ])
+
+        result = []
+        for k, v in time_dict.items():
+            if v == 1:
+                result.append('%s %s' % (v, k))
+            elif v > 1:
+                result.append('%s %ss' % (v, k))
+
+        return ' '.join(result)
+
+    def check_request_state(self, abs_diff, rel_diff, state_, value, duration):
+        """
+        Checks if the review request is older or newer than specified
+        time interval.
+        Args:
+            abs_diff (relativedelta): absolute time difference between now and
+                                      review request filed
+
+            rel_diff (str): relative time difference between now and
```

Comment:
My bad! They both are relative-delta objects. 
Example depicting the distinction:
time_a = 4:00 pm
time_b = 5:00 pm
rel_diff would be 1 hour, 0 minute, 0 seconds
abs_diff would be 1 hour or 60 minutes or 3600 seconds.